bemserver_core/database.py

```diff
@@ -174,3 +175,13 @@ def create_all(self, **kwargs):
 
 
 db = DBConnection()
+
+
+# Inspired from https://stackoverflow.com/a/36732359
+@sqla.event.listens_for(DB_SESSION, "before_flush")
+def receive_before_flush(session, flush_context, instances):
+    """Listen for the `before_flush` event"""
+    # Try to call _before_flush method for each modified object instance in session.
+    for obj in chain(session.new, session.dirty):
+        if hasattr(obj, "_before_flush"):
```

Comment:
Or we add an empty `_before_flush` to the `Base` class. Then no need to test and the model is more explicit.

Reply:
Right, it is better like that.